### PR TITLE
Fix: #19 going with localStorage sync, which was not done properly

### PR DESCRIPTION
**Pull request by mehul1011**
Future tasks could be more cleanup on code for syncing the localStore, ie making it more modular.

and implementation of NgRx store upto date with localStore (i have to disable some of it's code for now)

please create these tasks as an improvement as required.

### CODE REVIEW

File: src/app/chat/chat-content/chat-content.component.ts [hunk 1]
```diff
@@ -78,8 +78,16 @@ export class ChatContentComponent
   }
 
   async createCompletion(element: HTMLTextAreaElement) {
+    const id = this.route.snapshot.paramMap.get('id');
+    console.log(this.messages);
+    // if (id) {
```
Comment (yagizhanNY) [inline src/app/chat/chat-content/chat-content.component.ts:83]: if these commented lines are not needed any more, please delete it.

File: src/app/chat/chat-content/chat-content.component.ts [hunk 2]
```diff
@@ -88,15 +96,20 @@ export class ChatContentComponent
...
-    this.store
-      .select((state) => state.data)
-      .subscribe((data) => {
-        this.messages = data;
-      });
-    this.scrollToBottom();
+
+    this.messages.push(message);
+
+    // this.store.dispatch(updateChatDataAction({ newChatMessage: message }));
```
Comment (yagizhanNY) [inline src/app/chat/chat-content/chat-content.component.ts:106]: if these commented lines are not needed any more, please delete it.

File: src/app/chat/chat-content/chat-content.component.ts [hunk 3]
```diff
@@ -110,23 +123,48 @@ export class ChatContentComponent
...
-        .subscribe((data) => {
-          this.messages = data;
-        });
-      this.chatDataService.setLocalStorageForAllChat(
-        this.currChatSelected,
-        this.messages
-      );
-      this.chatService.triggerEventForChatCreation(this.messages);
+      this.messages.push(responseMessage);
+      // this.store.dispatch(
```
Comment (yagizhanNY) [inline src/app/chat/chat-content/chat-content.component.ts:132]: if these commented lines are not needed any more, please delete it.

File: src/app/sidebar/sidebar.component.ts [hunk 4]
```diff
@@ -67,14 +71,27 @@ export class SidebarComponent implements OnInit {
...
     this.keys.push(newChatKey);
-    const newChatDataArray: Message[] = [];
+    let newChatDataArray: Message[] = [];
     if (data) {
-      for (let i = 0; i < data?.length; i++) {
-        newChatDataArray.push(data[i]);
-      }
-
+      newChatDataArray = [...data];
+      // for (let i = 0; i < data?.length; i++) {
```
Comment (yagizhanNY) [inline src/app/sidebar/sidebar.component.ts:80]: if these commented lines are not needed any more, please delete it.

**Review by yagizhanNY — CHANGES_REQUESTED** [review]

File: src/app/chat/chat-content/chat-content.component.ts (hunk 1 above)
Comment (mehul1011) [reply src/app/chat/chat-content/chat-content.component.ts:83]: For state management i need those later on. Is it fine to keep it, maybe we can have an issue as a reminder for deleting later.

Let me know in any case, i can keep it in my local and push in the branch without it.

**Comment by mehul1011** [comment]
For state management i need those later on. Is it fine to keep it, maybe we can have an issue as a reminder for deleting later.

Let me know in any case, i can keep it in my local and push in the branch without it.

**Comment by yagizhanNY** [comment]
> For state management i need those later on. Is it fine to keep it, maybe we can have an issue as a reminder for deleting later.
> 
> Let me know in any case, i can keep it in my local and push in the branch without it.

Yes, it would be good if it's possible.

**Comment by mehul1011** [comment]
Hi @yagizhanNY 
I have removed it, let's merge, we can stabilize the UI

**Review by yagizhanNY — APPROVED** [review]

**Pull request merged**